eng/validation/update-channel.ps1

```diff
@@ -24,65 +26,118 @@ function Get-AzDO-Build([string]$token, [int]$azdoBuildId) {
     return $content | ConvertFrom-Json
 }
 
+function Get-AzDOHeaders()
+{
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    return $headers
+}
+
+function Get-LatestBuildResult([PSObject]$repoData)
+{
+    ## Verified that this API gets completed builds, not in progress builds
+    $headers = Get-AzDOHeaders
+    $uri = "https://dev.azure.com/$($repoData.azdoOrg)/$($repoData.azdoProject)/_apis/build/latest/$($repoData.buildDefinitionId)?branchName=$($repoData.subscribedBranchName)&api-version=5.1-preview.1"
+    $response = (Invoke-WebRequest -Uri $uri -Headers $headers -Method Get) | ConvertFrom-Json
+
+    ## Report non-green repos for investigation purposes. 
+    if(($response.result -ne "succeeded") -and ($response.result -ne "partiallySucceeded"))
+    {
+        Write-PipelineTaskError -message "The latest build on '$($repoData.subscribedBranchName)' branch for the '$($repoData.githubRepoName)' repository was not successful." -type "warning"
+        Write-Host "##vso[task.complete result=SucceededWithIssues;]"
```

Comment:
[nit] 

This will only matter if we try to move this function to a common place where other scripts would use it, but  I wonder if it makes sense to move this logging command to the caller instead of this function after it sees that the return value was a false. It seems like an unexpected side effect that calling a Get function would mark my build as succeeding with issues.

If we move it, it would also mean we only send the logging command once instead of N times for each repo that isn't gree

Reply:
Good point. I moved it out to the else case in the conditional that checks the results of the bellwethers.